packages/@aws-cdk/core/lib/construct.ts

```diff
@@ -202,6 +203,10 @@ export class ConstructNode {
    * @returns a construct or undefined if there is no default child
    */
   public get defaultChild(): IConstruct | undefined {
+    if (this._defaultChild !== undefined) {
+      return this._defaultChild;
+    }
+
     const resourceChild = this.tryFindChild('Resource');
```

Comment:
Do you want to cache the default child in the case where default child is calculated?

Reply:
Not really. It would be a behavior change (depending on when you call the method) and that's not the point.

Reply:
okay, got it.